core/pathresolver.go

```diff
@@ -16,11 +18,18 @@ var ErrTooManyLinks = errors.New("exceeded maximum number of links in ipns entry
 // Resolves the given path by parsing out /ipns/ entries and then going
 // through the /ipfs/ entries and returning the final merkledage node.
 // Effectively enables /ipns/ in CLI commands.
-func Resolve(n *IpfsNode, p path.Path) (*merkledag.Node, error) {
-	return resolveRecurse(n, p, 0)
+func Resolve(ctx context.Context, n *IpfsNode, p path.Path) (*merkledag.Node, error) {
+	r := resolver{ctx, n, p}
```

Comment:
i think the given `ctx` needs to also take into account `n.Context()`, otherwise the node can be cancelled, and the resolution could try to keep going :( -- see https://godoc.org/github.com/jbenet/go-context/ext#WithParents for handling this sort of thing.